src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -91,22 +95,33 @@ public void onPost(String triggeredByUser) {
      * Called when a POST is made.
      */
     public void onPost(final GitHubTriggerEvent event) {
+        if (Objects.isNull(job)) {
+            return; // nothing to do
+        }
+
+        Job<?, ?> currentJob = notNull(job, "Job can't be null");
+
         final String pushBy = event.getTriggeredByUser();
         DescriptorImpl d = getDescriptor();
         d.checkThreadPoolSizeAndUpdateIfNecessary();
         d.queue.execute(new Runnable() {
             private boolean runPolling() {
                 try {
-                    StreamTaskListener listener = new StreamTaskListener(getLogFile());
+                    StreamTaskListener listener = new StreamTaskListener(getLogFileForJob(currentJob));
 
                     try {
                         PrintStream logger = listener.getLogger();
+
                         long start = System.currentTimeMillis();
                         logger.println("Started on " + DateFormat.getDateTimeInstance().format(new Date()));
                         if (event.getOrigin() != null) {
                             logger.format("Started by event from %s on %tc%n", event.getOrigin(), event.getTimestamp());
                         }
-                        boolean result = SCMTriggerItems.asSCMTriggerItem(job).poll(listener).hasChanges();
+                        SCMTriggerItem item = SCMTriggerItems.asSCMTriggerItem(currentJob);
+                        if (null == item) {
+                            throw new RuntimeException("Job is not an SCMTriggerItem: " + currentJob);
```

Comment:
Could we please avoid Runtime Exception and throw a proper exception?

Like illegal state exception?

Reply:
I threw a runtime exception, because the code already had a catch for it below/didn't want to make a signature change.

Reply:
Illegal state exception is also a runtime exception just more descriptive